timeboost-sequencer/src/decrypt.rs

```diff
@@ -204,36 +223,67 @@ impl Decrypter {
             }
         }
 
-        // normal loop: try to receive the next decrypted inclusion list
-        while let Some(dec_incl) = self.worker_rx.recv().await {
-            let round = dec_incl.round();
-            // update decrypter cache of inclusion list
-            debug!(
-                node  = %self.label,
-                round = %round,
-                epoch = %dec_incl.epoch(),
-                "inclusion list decrypted"
-            );
-            debug_assert!(
-                !dec_incl.is_encrypted(),
-                "decrypter worker returns non-decrypted inclusion list"
-            );
-            self.incls.insert(round, Status::Decrypted(dec_incl));
-
-            // since the newly finished/responded inclusion list might belong to a later round
-            // the first entry might still be unencrypted, in which case continue the loop and
-            // wait until the worker finishes decrypting it.
-            if let Some(entry) = self.incls.first_entry() {
-                if matches!(entry.get(), Status::Decrypted(_)) {
-                    if let Status::Decrypted(incl) = entry.remove() {
-                        return Ok(incl);
+        loop {
+            tokio::select! {
+                end = &mut self.worker => match end {
+                    Ok(end) => {
+                        let end: EndOfPlay = end;
+                        error!(node = %self.label, %end, "worker terminated");
+                        return Err(DecrypterDown(()));
+                    }
+                    Err(err) => {
+                        error!(node = %self.label, %err, "worker panic");
+                        return Err(DecrypterDown(()));
+                    }
+                },
+                msg = self.worker_rx.recv() => {
+                    match msg {
```

Comment:
```suggestion
                msg = self.worker_rx.recv() => match msg 
```
nit: one less indentation